### ISSUE TITLE
ID integer values too large for column type 'integer' in postgres

### ISSUE THREAD

**Issue by ccwittwer** [CodeSleeve/fixture#3]
generateKey function in Repository outputs a 10-digit integer, which is used for a record ID.  The integer can exceed max size for Postgres integer column.
```php
	protected function generateKey($value)
	{
		$hash = sha1($value);
		$integerHash = base_convert($hash, 16, 10);

		return (int)substr($integerHash, 0, 10);
	}
```

Changing the substring length will correct the issue.
```php
	protected function generateKey($value)
	{
		$hash = sha1($value);
		$integerHash = base_convert($hash, 16, 10);

		return (int)substr($integerHash, 0, 8);
	}
```

**Comment by clphillips** [comment]
generateKey() always returns the max 32-bit signed integer when using 32-bit PHP.

**Comment by b3njamin** [comment]
This is fixed in #12. With the ability to add custom key generators.
